tests/generators/test_huggingface.py

```diff
@@ -1,11 +1,25 @@
 import transformers
 import garak.generators.huggingface
+from garak._config import GarakSubConfig
 
 DEFAULT_GENERATIONS_QTY = 10
 
 
 def test_pipeline():
-    g = garak.generators.huggingface.Pipeline("gpt2")
+    gen_config = {
+        "huggingface": {
+            "Pipeline": {
+                "name": "gpt2",
+                "hf_args": {
+                    "device": "cpu",
+                },
+            }
+        }
+    }
+    config_root = GarakSubConfig()
+    setattr(config_root, "generators", gen_config)
+
+    g = garak.generators.huggingface.Pipeline("gpt2", config_root=config_root)
```

Comment:
side note: this is a working but i guess relatively inelegant pattern for getting a generator up. 

1. What happens if the config setting is skipped? 
2. Is the name parameter in `Pipeline`'s constructor redundant? How does it relate to the `name` in `gen_config` - which takes precedence?
3. why device `cpu` and not `auto`?

Reply:
Good questions:
1. If config is skipped the new auto detection will load the most efficient device available. On macOS this turns out to be `mps` and actually causes this class to fail construction due to incomplete support for MPS in torch at this time.
2. `name` duplication is redundant, currently `name` from the constructor is help over `name` in `config_root` however either could be removed for this test.
3. See answer to `1`, `cpu` is a guaranteed fully supported value.

Reply:
OK, that's good to know.

1. I think we probably don't want to ship logic that breaks generators by default on macOS - did I read this right?
2. Sounds like there is some tidying up to do. I would love for the default config to run reasonably well out of the box, especially with `Pipeline`+`gpt2`, which is used in so many examples because it has a high chance of fitting in memory and going quickly on any machine. Requiring a `setattr()` gives an impression of "hotwiring". I'll think on this more - I haven't had time to play with the #711 config updates in depth much.
3. OK, that's cool :)

Reply:
1. `mps` does not break on macOS by default as `gpt2` does work,  I have tested it will load and complete a `fast` config, however this test needs to be consistent for all runs so forcing `cpu` was a reasonable solution. This test in particular results in the error below when asking for generation on `""` on `mps`, the error seemed like a clear message to the user since overriding the device can be a straight forward config setting now.
```
ERROR    root:huggingface.py:203 Can't infer missing attention mask on `mps` device. Please provide an `attention_mask` or use a different device.
```
2. The `setattr()` here could be replaced with a simple dictionary as both are supported by config I just followed more closely what `cli` will generate for as that is the more specific object type.
``` Python
config_root = {
        "generators": {    
            "huggingface": {
                "Pipeline": {
                    "name": "gpt2",
                    "hf_args": {
                        "device": "cpu",
                    },
                }
            }
        }
    }
```

Reply:
We don't test all generators for empty string behavior by default because they each their own unique special snowflake style (making it a schlep) and often require an API key (making it an expensive pain). I tried to write this one day and discovered it was slightly less feasible than imagined - the new config setup might make it easier now. There are a few tests in place, but it's incomplete.